src/passes/OptimizeInstructions.cpp

```diff
@@ -1265,10 +1285,22 @@ struct OptimizeInstructions
   // but it's still worth doing since
   //  * Often shifts are more common than muls.
   //  * The constant is smaller.
-  Expression* optimizePowerOf2Mul(Binary* binary, uint32_t c) {
-    uint32_t shifts = CountTrailingZeroes(c);
-    binary->op = ShlInt32;
-    binary->right->cast<Const>()->value = Literal(int32_t(shifts));
+  template<typename T> Expression* optimizePowerOf2Mul(Binary* binary, T c) {
+    int32_t shifts = CountTrailingZeroes<T>(c);
+    binary->op = sizeof(T) <= 4 ? ShlInt32 : ShlInt64;
+    binary->right->cast<Const>()->value = Literal(static_cast<T>(shifts));
+    return binary;
+  }
+
+  // Optimize an unsigned divide by a power of two on the right
```

Comment:
this comment could say "divide / remainder" and then we can remove the duplicated comment on the later function